File: leetcode/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */
package main

// @lc code=start
func twoSum(nums []int, target int) []int {
	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}

// @lc code=end

File: leetcode/1.two-sum.scala
/*
 * @lc app=leetcode id=1 lang=scala
 *
 * [1] Two Sum
 */

// @lc code=start
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val map = nums.zipWithIndex.foldLeft(Map[Int, Int]()) { case (acc, (v, i)) => acc + (v -> i)}
        for ((v, i) <- nums.zipWithIndex) {
            map.get(target - v) match {
                case None => ();
                case Some(x) => if (i != x) return Array(i, x) else ();
            }
        }
        return nums;
    }
}
// @lc code=end

